src/pages/suggestions/new/index.tsx

```diff
@@ -1,92 +1,101 @@
+  initialState,
+  setActiveArea,
+} from "@/redux/slices/new-suggestion.slice"
+import { insertSuggestion, insertSuggestionInstruments } from "@/services/suggestion.service"
 import { Database } from "@/types/database"
-import { Instrument } from "@/types/database-types"
-import { useSupabaseClient } from "@supabase/auth-helpers-react"
-import { getInstruments } from "@/services/instrument.service"
-import Spinner from "@/components/utils/spinner"
-import ErrorPopup from "@/components/popups/error-popup"
+import { useSupabaseClient, useUser } from "@supabase/auth-helpers-react"
+import { useRouter } from "next/router"
+import { mapInstruments } from "@/helpers/new-suggestion.helper"
+import { InputsSongInformation, ISuggestionInstrument } from "@/interfaces/suggestion"
 
 const NewSuggestion = () => {
-  const router = useRouter()
-  const activeArea = useSelector((state: AppState) => state.newSuggestion.activeArea)
   const suggestion = useSelector((state: AppState) => state.newSuggestion.suggestion)
+  const activeArea = useSelector((state: AppState) => state.newSuggestion.activeArea)
 
-  const methods = useForm<InputsSongInformation>({
-    defaultValues: { ...suggestion, artist: suggestion.artist.join(",") } as InputsSongInformation,
-    shouldFocusError: false,
-  })
-
-  const supabaseClient = useSupabaseClient<Database>()
-  const [instrumentList, setInstrumentList] = useState<Instrument[]>([])
-  const [showSpinner, setShowSpinner] = useState<boolean>(false)
-  const [showLoadingError, setShowLoadingError] = useState<boolean>(false)
+  const supabase = useSupabaseClient<Database>()
+  const dispatch = useDispatch()
+  const router = useRouter()
+  const user = useUser()
 
-  useEffect(() => {
-    setShowSpinner(true)
-    getInstruments(supabaseClient)
+  const saveSuggestion = (onSuccess: () => void, onError: () => void) => {
+    if (user === null) {
+      onError()
+      return
+    }
+    insertSuggestion(supabase, suggestion, user.id)
       .then((response) => {
-        if (response.error || response.data?.length === 0) {
-          setShowLoadingError(true)
+        if (response.error) {
+          onError()
           return
         }
 
-        setInstrumentList(response.data as Instrument[])
+        const suggestionId = response.data.at(0)!.id
+        insertSuggestionInstruments(supabase, mapInstruments(suggestion, suggestionId))
+          .then((response) => {
+            if (response.error) {
+              onError()
+              return
+            }
+
+            router.push("/suggestions").then(() => {
+              onSuccess()
+              dispatch(updateNewSuggestion(initialState.suggestion))
+              dispatch(setActiveArea(Area.SongInformation))
+            })
+          })
+          .catch(() => onError())
       })
-      .catch(() => {
-        setShowLoadingError(true)
+      .catch(() => onError())
+  }
+
+  const onAreaSelect = (area: Area) => {
+    dispatch(setActiveArea(area))
+  }
```

Comment:
Could be written as a one liner but is okay if it's not :)

Reply:
Prettier automaticaly changes it to this structure.

Reply:
Try removing the curly brackets first